Chapter10/pretty_print.erl

```diff
@@ -136,59 +127,51 @@ pad(_, Word) -> Word.
-compress([Head,Next|Tail], Output) ->
-	if
-		(Next - Head) == 0 -> compress([Head|Tail], Output);
-		% (Next - Head) == 1 -> compress([Next|Tail], Output ++ [Head] ++ ["-"] );
-		(Next - Head) >= 1 -> compress([Next|Tail], Output ++ [Head])
-	end.
-
-
 %%Linker (Links adjacent numbers) [1,2,5,8,9,10] -> [{1,2},{5},{8,10}]
 compress_link([]) -> [];
 compress_link(List) -> compress_link(List, {0, 0}, []).
 
-compress_link([], _, Output) -> Output;
-compress_link([Head,Next], Range, Output) -> 
-	if
-		(Next - Head == 1) and (Range == {0, 0}) -> Output ++ [{Head, Next}];
-		(Next - Head == 1) -> Output ++ [{element(1, Range), Next}];
-		(Next - Head >= 1) and (Range /= {0, 0}) -> Output ++ [{element(1, Range),Head},{Next}];
-		(Next - Head >= 1) -> Output ++ [{Head},{Next}]
-	end;
-compress_link([Head,Next|Tail], Range, Output) ->
-	if
-		(Next - Head == 1) and (Range == {0, 0}) -> compress_link([Next|Tail], {Head, 0}, Output);
-		(Next - Head == 1) -> compress_link([Next|Tail], {element(1, Range), Next}, Output);
-		(Next - Head >= 1) and (Range == {0, 0}) -> compress_link([Next|Tail], {0, 0}, Output ++ [{Head}]);
-		(Next - Head >= 1) -> compress_link([Next|Tail], {0, 0}, Output ++ [{element(1, Range),Head}])
-	end.
-
-
-%Prettyer (List with tuples to string) [{1,2},{3},{5,9}] -> "1-2,3,5-9,"
-compress_pretty([]) -> [];
-compress_pretty([Head|Tail]) when tuple_size(Head) == 1 -> char_to_integer(element(1,Head)) ++ "," ++ compress_pretty(Tail);
-compress_pretty([Head|Tail]) when tuple_size(Head) == 2 ->
-	char_to_integer(  element(1,Head)) ++ "-" ++ char_to_integer(element(2,Head)  )
-	++ "," ++ compress_pretty(Tail).
-
-
-char_to_integer(Char) -> lists:flatten(io_lib:format("~p", [Char])).
-
-
+compress_link([], _Range, Output) -> Output;
 
+compress_link([Head,Next], {0, 0}, Output)
+  when (Next - Head == 1) -> Output ++ [{Head, Next}];
+compress_link([Head,Next], Range, Output)
+  when (Next - Head == 1) -> Output ++ [{element(1, Range), Next}];
+compress_link([Head,Next], Range, Output)
+  when (Next - Head >= 1) and (Range /= {0, 0}) ->
+    Output ++ [{element(1, Range),Head},{Next}];
+compress_link([Head,Next], _Range, Output)
+  when (Next - Head >= 1) -> Output ++ [{Head},{Next}];
 
 
+compress_link([Head,Next|Tail], {0, 0}, Output)
+  when (Next - Head == 1) -> compress_link([Next|Tail], {Head, 0}, Output);
+compress_link([Head,Next|Tail], Range, Output)
+  when (Next - Head == 1) -> compress_link([Next|Tail], {element(1, Range), Next}, Output);
+compress_link([Head,Next|Tail], {0,0}, Output)
+  when (Next - Head >= 1) -> compress_link([Next|Tail], {0, 0}, Output ++ [{Head}]);
+compress_link([Head,Next|Tail], Range, Output)
+	when (Next - Head >= 1) -> compress_link([Next|Tail], {0, 0}, Output ++ [{element(1, Range),Head}]).
 
 
+%Prettyer (List with tuples to string) [{1,2},{3},{5,9}] -> "1-2,3,5-9"
+compress_pretty([]) -> [];
 
+compress_pretty([Head = {_Entry}|Tail]) ->
```

Comment:
why didn't you use `Entry` instead of `element(1, Head)`?

Reply:
Removed all use of "element/2", now instead pattern match to variable